community/io/src/main/java/org/neo4j/io/pagecache/impl/SingleFilePageSwapper.java

```diff
@@ -500,7 +503,19 @@ private long lockPositionWriteVector(
             synchronized ( positionLock( channel ) )
             {
                 channel.position( fileOffset );
-                return channel.write( srcs );
+
+                /*
+                 * So the problem here is that calling channel.write( ByteBuffer[] ) sometimes doesn't work as expected
+                 * and writes less data than what is asked to write into the file. Changing the code by calling
+                 * channel.write( ByteBuffer ) in a loop looks like it is working in the expected way.
+                 */
+                long written = 0;
+                for ( ByteBuffer src : srcs )
+                {
+                    written += channel.write( src );
+                }
+
+                return written;
```

Comment:
This is not vectored IO, though. The whole point of this exercise is to use the `channel.write( ByteBuffer[] )` method. Let's talk offline.